skbio/stats/composition.py

```diff
@@ -608,10 +611,329 @@ def centralize(mat):
+    we default to ``scipy.stats.f_oneway`` here, which can be used when
+    there are two or more groups. Users should refer to the documentation
+    of these tests in SciPy to understand the assumptions made by each test.
+
+    This method cannot handle any zero counts as input, since the logarithm
+    of zero cannot be computed.  While this is an unsolved problem, many
+    studies have shown promising results by replacing the zeros with pseudo
+    counts.  This can be also be done via the multiplicative_replacement method
+
+
+    References
+    ----------
+    .. [1] Mandal et al. "Analysis of composition of microbiomes: a novel
+       method for studying microbial composition", Microbial Ecology in Health
+       & Disease, (2015), 26
+
+    Examples
+    --------
+    First import all of the necessary modules:
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now let's load in a pd.DataFrame a count matrix with 6 samples
+    and 7 unknown bacteria:
+
+    >>> table = pd.DataFrame([[12, 11, 10, 10, 10, 10, 10],
+    ...                       [9,  11, 12, 10, 10, 10, 10],
+    ...                       [1,  11, 10, 11, 10, 5,  9],
+    ...                       [22, 21, 9,  10, 10, 10, 10],
+    ...                       [20, 22, 10, 10, 13, 10, 10],
+    ...                       [23, 21, 14, 10, 10, 10, 10]],
+    ...                      index=['s1','s2','s3','s4','s5','s6'],
+    ...                      columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a grouping vector.  In this scenario, there
+    are only two classes, and suppose these classes correspond the
+    treatment due to a drug and a control.  The first three samples
+    are controls and the last three samples are treatments.
+
+    >>> grouping = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                      index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run `ancom` and see if there are any features that have any
+    significant differences between the treatment and the control.
+
+    >>> results = ancom(table, grouping)
+    >>> results['W']
+    b1    0
+    b2    4
+    b3    1
+    b4    1
+    b5    1
+    b6    0
+    b7    1
+    Name: W, dtype: int64
+
+    The W-statistic is the number of features that a single feature is tested
+    to be significantly different against. In this scenario, b2 was detected to
+    have significantly different abundances compared four of the other species.
+    To summarize the results from the W-statistic, let's take a look and the
+    results from the hypothesis test:
+
+    >>> results['reject']
+    b1    False
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+
+    From this we can conclude that only b2 was significantly
+    different between the treatment and the control.
+
+    """
+    if len(table) != len(grouping):
```

Comment:
In addition to having the same length, the index on both objects needs to be the same. The two objects should be aligned so that order doesn't matter.

I think the fastest way is to do this is storing the number of samples in `table` and `grouping`, then calling `table.align(grouping, axis=0, join='inner')`, which will produce an aligned DataFrame and Series, then checking if the number of samples in the aligned objects changed from the originals. If they changed, raise an error. Does that make sense?